metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/ThumbnailGenerator.java

```diff
@@ -371,12 +372,18 @@ ImageMetadata parseCommandResponse(List<String> response) throws MediaExtraction
 
   private static List<String> extractDominantColors(List<String> results, int skipLines) {
     final Pattern pattern = Pattern.compile("#([0-9A-F]{6})");
+
+    final boolean aResultDoesNotMatch = results.stream().skip(skipLines)
+        .sorted(Collections.reverseOrder()).limit(COMMAND_RESULT_MAX_COLORS).map(pattern::matcher)
+        .anyMatch(m -> !m.find());
```

Comment:
This seems to create duplicate code. Maybe better is to define a stream variable and then reuse it? And also, you can make the anyMatch check nicer, I think? 

How about this?

```
final Stream<Matcher> matchers =  results.stream().skip(skipLines)
        .sorted(Collections.reverseOrder()).limit(COMMAND_RESULT_MAX_COLORS).map(pattern::matcher);
if (!matchers.allMatch(matcher::find) {
  throw new IllegalStateException("Invalid color line found.");
}
return matchers.map(matcher -> matcher.group(1)).collect(Collectors.toList());
```